test/oauth.js

```diff
@@ -23,41 +22,87 @@ describe('oauth', function() {
+      clientSecret: 'fake_client_secret',
+      redirectUri: 'some-redirect-uri',
+    }
+    const expectedResponse = {
+      refresh_token: 'a_fake_refresh_token',
+      access_token: 'a_fake_access_token',
+      expires_in: 21600,
+    }
+    const oauthTokenEndpoint = {
+      path: '/oauth/v1/token',
+      request: {
+        grant_type: 'authorization_code',
+        client_id: clientProperties.clientId,
+        client_secret: clientProperties.clientSecret,
+        redirect_uri: clientProperties.redirectUri,
+        code,
+      },
+      response: expectedResponse,
+    }
+    fakeHubspotApi.setupServer({ postEndpoints: [oauthTokenEndpoint] })
+
+    if (process.env.NOCK_OFF) {
+      it('should return a token from hubspot')
+    } else {
+      it('should return a token from hubspot', function() {
+        if (process.env.NOCK_OFF) {
+          console.warn(
+            'we currently do not test oauth with NOCK_OFF set to true',
+          )
+          return
+        }
+
+        hubspot = new Hubspot(clientProperties)
+
+        return hubspot.oauth.getAccessToken({ code }).then(data => {
+          expect(data).to.deep.equal(expectedResponse)
+        })
       })
-    })
+    }
+  })
 
-    it('should return (and refresh) an accessToken, given a refreshToken - params', function() {
-      const hubspot = new Hubspot({ accessToken: process.env.accessToken })
-      if (!process.env.refreshToken) {
-        return
-      } // hard to reproduce on CI. local testing only for now
-      const params = {
-        client_id: process.env.clientId,
-        client_secret: process.env.clientSecret,
-        redirect_uri: process.env.redirectUri,
-        refresh_token: process.env.refreshToken,
-      }
-      return hubspot.oauth.refreshAccessToken(params).then(res => {
-        expect(res.refresh_token).to.equal(params.refresh_token)
-        expect(res).to.have.a.property('access_token')
-        expect(hubspot.auth.bearer).to.equal(res.access_token)
+  describe('refreshAccessToken', function() {
+    const clientProperties = {
+      clientId: 'fake_client_id',
+      clientSecret: 'fake_client_secret',
+      redirectUri: 'some-redirect-uri',
+      refreshToken: 'a_fake_refresh_token',
+      accessToken: 'a_fake_access_token',
+    }
+    const expectedResponse = {
+      refresh_token: 'a_new_fake_refresh_token',
+      access_token: 'a_new_fake_access_token',
+      expires_in: 21600,
+    }
+    const oauthTokenEndpoint = {
+      path: '/oauth/v1/token',
+      request: {
+        grant_type: 'refresh_token',
+        client_id: clientProperties.clientId,
+        client_secret: clientProperties.clientSecret,
+        redirect_uri: clientProperties.redirectUri,
+        refresh_token: clientProperties.refreshToken,
+      },
+      response: expectedResponse,
+    }
+    fakeHubspotApi.setupServer({ postEndpoints: [oauthTokenEndpoint] })
+
+    if (process.env.NOCK_OFF) {
+      it('should return a token from hubspot')
```

Comment:
Thoughts on adding a similar warning here?

Reply:
Turns out it doesn't look great